### ISSUE TITLE
CSV Import from our Documenation doesn't work. 

### ISSUE THREAD

**Issue by zdenek-jonas** [microstream-one/microstream#71]
I have tried to perform import exactly as described in our Getting started documentation.
https://manual.docs.microstream.one/data-store/import-export

I have finished with error message:
```one.microstream.exceptions.IORuntimeException: java.io.IOException: Der Prozess kann nicht auf die Datei zugreifen, da ein anderer Prozess einen Teil der Datei gesperrt hat```

I have tried to copy exported CSV Files into separate Folder and started the import with the same error again. 

I see this problem as critical because this functionality is in our Documentation. 

Test:
```
    @Test
    public void csvReadTest() {

        EmbeddedStorageManager reimportStorage = EmbeddedStorage.start();

        File binary = new File("c://devel//tmp/binary/");

        StorageDataConverterTypeCsvToBinary<StorageFile> reimportConverter =
                StorageDataConverterTypeCsvToBinary.New(
                        StorageDataConverterCsvConfiguration.defaultConfiguration(),
                        reimportStorage.typeDictionary(),
                        new StorageEntityTypeConversionFileProvider.Default(binary.toPath(), "dat")
                );

        File directory = new File("c://devel//tmp/csv/");
        List<File> importFiles = (List<File>) FileUtils.listFiles(directory, null, true);

        for (File file : importFiles) {
            StorageLockedFile storageFile = StorageLockedFile.openLockedFile(file.toPath());
            try {
                reimportConverter.convertCsv(storageFile);
            } finally {
                storageFile.close();
            }
        }
    }
```

callstack:
```
one.microstream.exceptions.IORuntimeException: java.io.IOException: Der Prozess kann nicht auf die Datei zugreifen, da ein anderer Prozess einen Teil der Datei gesperrt hat
at one.microstream.storage.types.StorageDataConverterTypeCsvToBinary$Default.parseCurrentFile(StorageDataConverterTypeCsvToBinary.java:1085)
at one.microstream.storage.types.StorageDataConverterTypeCsvToBinary$Default.convertCsv(StorageDataConverterTypeCsvToBinary.java:1703)
[...]
```

**Comment by tm-ms** [comment]
Probably the same problem as microstream-one/microstream-private#231.

**Comment by tm-ms** [comment]
Fixing microstream-one/microstream-private#231 did surprisingly not fix this issue as well.
So I did some additional testing and research:
`MainTestReadLockedFile` shows:
Reading from the locked file via the FileChannel that created the lock works.
Doing exactely the same with a newly opened FileChannel still produces the exception.
(Funny enough, opening the channel does NOT cause the exception, which I would consider inconsistent to the reading behavior, but whatever)

This means:
The file locking as implemented in Java NIO has nothing to do with the process, but with the "owner" channel of the lock. Not the process locks the file, but one particular FileChannel instance locks the file and no other FileChannel instance can read from that file.
BUT: another FileChannel instance CAN very well be used to copy the content of the file (which is, again, inconsistent).
AND: the old streams completely circumvent that mechanism and read from whereever they want despite any locks (and again, inconsistent behavior)

Bottom line: NIO is crap when it comes to locking.

However, I think I found a pretty simple solution for this issue:
The Csv Converter knows the StorageFile instance and that knows the FileLock "owning" FileChannel. So instead of creating a new FileChannel on the file defined by the current StorageFile instance, its original channel instance is used directly.
I wonder a little why I didn't do that in the first place. Probably lost in refactoring and as long as the redundant FileChannel worked, there was no need to change anything.
Now there is.
So I did.

According to the MainTest file, this should be enough.

**Comment by zdenek-jonas** [comment]
Ok, ich kann ein Schritt weiter gehen, dann kommt aber weitere Exception:

```
one.microstream.storage.exceptions.StorageException: Type not found: java.lang.String_30
at one.microstream.storage.types.StorageDataConverterTypeCsvToBinary$Default.beginTable(StorageDataConverterTypeCsvToBinary.java:1729)
at one.microstream.chars.XCsvParserCharArray.parseSegment(XCsvParserCharArray.java:837)
[...]
```

**Comment by tm-ms** [comment]
Since legacy type mapping, the type name alone is no longer sufficient to identify a type (and its version). To create unique and reasonably named export file names, I added the typeId after the type name.
However, I complete completely neglected the reverse way of convering csv to binary and importing binary files. So the csv to binary conversion logic still tries to parse the whole file name as the type name.

To fix this, I changed the parsing logic to extract the typeId and use that to resolve the actual type.

**Comment by zdenek-jonas** [comment]
Works... getestet
